content/en/blog/2024/otel-errors/index.md

```diff
@@ -0,0 +1,286 @@
+After all this, you might be wondering which signal to use to capture errors: 
+spans or logs? The answer is: "It depends!" Perhaps your team primarily uses 
+traces; perhaps they primarily use logs.
+
+Spans can be great for capturing errors, because if the operation errors out, 
+marking a span as an error makes it stand out and therefore easier to spot. On 
+the other hand, if you’re not filtering or tail sampling your traces and your 
+system is producing thousands of spans per minute, you could miss errors that 
+aren’t occurring frequently, but that still need to be handled. 
+
+What about using span events versus logs? Again, this depends. It may be 
+convenient to use span events, because when a span status is set to `Error`, a 
+span event with the exception message (and other metadata you may wish to capture) 
+is automagically created. 
+
+Another consideration is your observability backend. Does your backend render 
+both logs and traces? How easily queryable or discoverable are your logs, spans, 
+and span events? Is logs and trace correlation supported?
+
+## Visualizing Errors in Different Backends:
+While OTel provides us with the raw telemetry data emitted by our systems, it 
+doesn’t provide data visualization or interpretation. This is done by an 
+observability backend. Because OTel is vendor-neutral, it means that the same 
+information emitted can be visualized and interpreted by different backends 
+without re-instrumenting your application. 
+
+If you have been using a proprietary agent to monitor your applications and have 
+recently migrated to OTel, you might notice that an OTel error may not be 
+expressed the way you expect in your observability backend, as compared to the 
+same error captured by the proprietary agent. This is  most likely due to the fact 
+that OTel simply models errors differently than how vendors have been modeling 
+them. 
+
+As one example, OTel’s notion of span kinds may affect how your OTel error is 
+visualized. For instance, if you have a trace that has one exception and it’s on 
+an internal span with its status set to `Error`, you should see the trace marked 
+with an error, but it may not be counted toward your app error rate. This is 
+because the backend may have an opinion that only errors on entry point spans 
+(server spans) and consumer spans should be counted toward your error rate. 
+
+Furthermore, while [Jaeger](https://www.jaegertracing.io/) visualizes span events 
+as logs, some backends may synthesize span events as its own data type instead of 
+as a log data type, which would impact the way you query that data. 
+
+What might some of these errors look like in an observability backend? We’ll take 
+a look at Jaeger and a couple proprietary examples. The following screen captures 
+show how different observability backends render the same error data differently. 
+The error data was generated by the code in [this repo](https://github.com/avillela/otel-errors-talk).
```

Comment:
```suggestion
The error data was generated by the code in [this repository](https://github.com/avillela/otel-errors-talk).
```

the repo is still private I guess?

Reply:
Oh yes! We will make that public, thank you for the call-out.